Assets/Scripts/Enemy/StateMachine.cs

```diff
@@ -5,13 +5,18 @@
 
 public class StateMachine : MonoBehaviour
 {
-
     [SerializeField] private State CurrentState;
```

Comment:
Private variables are prefixed with an underscore. Should be `_currentState`. Should go through and fix all instances for consistency.